dev/com.ibm.ws.security.acme/src/com/ibm/ws/security/acme/internal/AcmeHistoricalFileHelper.java

```diff
@@ -0,0 +1,152 @@
+			while ((line = br.readLine()) != null && !line.isEmpty()) {
+				lastLine=line;
+			}
+			br.close();
+			if (lastLine != null && !lastLine.isEmpty()) {
+				StringTokenizer tok = new StringTokenizer(lastLine);
+				if (tok.hasMoreTokens()) tok.nextToken();
+				if (tok.hasMoreTokens()) tok.nextToken();
+				if (tok.hasMoreTokens()) fileDirURI = tok.nextToken();
+			}
+		} catch (Exception e) {
+			Tr.error(tc, e.getMessage());
+			return true;
+		}
+		if (currentDirectoryURI.equals(fileDirURI)) {
+			return false;
+		}
+		return true;
+	}
+	
+	/**
+	 * This method will create the initial acme file in the servers/workarea
+	 * directory. It holds certificate and directoryURI information, eg.
+	 *  Date                Serial                   DirectoryURI                    Account URI
+     *  --------------------------------------------------------------------------------------------
+     *  20200509231118      6542743894787011570      https://localhost:33827/dir     https://localhost:33827/my-account/1
+	 */
+	private void createAcmeFile(WsLocationAdmin wslocation) {
+		File file = wslocation.getServerWorkareaResource("acme/acme.txt").asFile();
+		if (file.exists()) return;
+		file.getParentFile().mkdirs();
+		try {
+			file.createNewFile();
+			FileWriter fr;
+			fr = new FileWriter(file, false);
+			fr.write("Date" + spaceDelim + "Serial" + spaceDelim + "DirectoryURI" + spaceDelim + "Account URI" + "\n");
+			fr.write("-------------------------------------------------------------------------------------------------------------------------");
+	       	fr.close();
+		} catch (IOException e) {
+			Tr.error(tc, e.getMessage());
+		}
+	}
+
+	/**
+	 * Updates the historical ACME file.
+	 * @param certificate The current X509Certificate.
+	 * @param directoryURI The current directoryURI from config.
+	 * @param accountURI The current accountURI from config.
+	 * @throws AcmeCaException
+	 */
+	protected void updateAcmeFile(X509Certificate certificate, String directoryURI, String accountURI, WsLocationAdmin wslocation) throws AcmeCaException {
+		updateAcmeFile(null, certificate, directoryURI, accountURI, wslocation);
+	}
+
+	/**
+	 * Update the acme file with the original or updated certificate
+	 * and directoryURI information. Max file size is 10 lines.
+	 * @param acmeCertificate The certificate to use if the default X509 certificate is null.
+	 * @param certificate The current X509Certificate.
+	 * @param directoryURI The current directoryURI from config.
+	 * @param accountURI The current accountURI from config.
+	 * @throws AcmeCaException
+	 */
+	protected void updateAcmeFile(AcmeCertificate acmeCertificate, X509Certificate certificate, String directoryURI, String accountURI, WsLocationAdmin wslocation) throws AcmeCaException {
+		File file = wslocation.getServerWorkareaResource("acme/acme.txt").asFile();
+		if (!file.exists()) {
+			createAcmeFile(wslocation);
+		}
+		//If the file still doesn't exist, return
+		if (!file.exists()) {
```

Comment:
Can't we return whether we created the file from createAcmeFile, then we wouldn't need to check if the file exists again?